lib/extension-core.js

```diff
@@ -96,4 +96,22 @@ function core(loader) {
 
     return Promise.resolve(loaderLocate.call(this, load));
   }
+
+  function applyExtensions(extensions, loader) {
+    loader._extensions = [];
+    for(var i = 0, len = extensions.length; i < len; i++) {
+      extensions[i](loader);
+    }
+  }
+
+  loader._extensions = [core];
```

Comment:
If we go with a more resilient pattern, then we can do:

```javascript
loader._extensions = (loader._extensions || []).push(core);
```

Which is then the same in every extension?